news/feature-3123.md

```diff
@@ -0,0 +1,70 @@
+
+The `python-http-header` plugin makes it possible for users to implement HTTP header plugins in Python language. It is built on top of signal-slot mechanism: currently HTTP module defines only one signal, that is `signal_http_header_request` and `python-http-header` plugin implements a python binding for this signal. This means that when the `signal_http_header_request` signal is emitted then the connected slot executes the Python code.
+
+The Python interface is :
+```
+def get_headers(self, body, headers):
+```
+
+It should return a tuple that has two elements:
+ * a timeout (when the signal is emitted again within `timeout`,  a cached list is returned - this can be used for caching Authorization header tokens that has an expiration time and as the check of the value of the timeout is done at C site, it can improve the performance: Python code runs only when it is required) 
+ * a string list (the headers that will be appended to the request's header)
+
+Code, or naming is not final, but it's working. Just wanted to open the PR and get some feedback (...and run CI tests).
+
+Original code was written by Ferenc Sipos.
+
+<details>
+  <summary>Example config, click to expand!</summary>
+
+```
+
+@version: 3.25
+
+python {
+from syslogng import Logger
+
+logger = Logger()
+
+class TestCounter():
+    def __init__(self, options):
+        self.header = options["header"]
+        self.counter = int(options["counter"])
+        self.timeout = 0
+        logger.debug(f"TestCounter class instantiated; options={options}")
+
+    def get_headers(self, body, headers):
+        logger.debug(f"get_headers() called, received body={body}, headers={headers}")
+       
+        response = (self.timeout, [f"{self.header}: {self.counter}"])
```

Comment:
I suggest the following change (also in the PR description).

`timeout` should not be part of self, it is intended to be calculated in `get_headers`.
`response` is not appropriate here. Probably `retval` was the intention. Still I would split timeout and headers, and write something similar, instead.

```
        headers = [f"{self.header}: {self.counter}"]
        cache_time = 0
        self.counter += 1
        return (cache_time, headers)
```

The `f` syntax is very neat in python3, but it is not available in python2. Until the python2 support is dropped, I would prefer to see a snippet that could be used in any environment. For example, use format, even though it is less readable.
```
        response = (self.timeout, ["{}: {}".format(self.header, self.counter)])
```

Reply:
Man, it's just an example... 
The one and only intention is to show how one can implement such a Python module.
It's an interesting question whether we should support python2 in our example modules... I can do that.